test/samplers/test_samplers.py

```diff
@@ -131,6 +134,80 @@ def test_sampling_range_negative(sampler):
         assert_tensor_equal(clip.data, clips_1[0].data)
 
 
+@pytest.mark.parametrize("sampler", (clips_at_random_indices, clips_at_regular_indices))
+def test_sampling_range_default_behavior(sampler):
```

Comment:
Okay, based on this comment, I think that explains the behavior I was confused about in [my comment above](https://github.com/pytorch/torchcodec/pull/241/files#r1790330006).

This means that effectively, the default values for the sampler are:
```
  sampling_range_start = 0,
  sampling_range_end = max(len(decoder) - clip_span + 1, 1),
```
I do think it may be simpler to just make them:
```
  sampling_range_start = 0,
  sampling_range_end = len(decoder),
```
And then let the policy deal with the case that we start a clip in the range `[len(encoder) - clip_size, len(encoder))`. (I may be off by 1 somewhere.) I don't have a strong preference for this, but I think it will make our code more straight-forward, and I think it's easier to explain the behavior. I'm unsure if it's actually a _better_ result for the user, though.